### PR TITLE
Add table_aws_kinesis_stream. closes #111

### LINKED ISSUE #111: Add table_aws_kinesis_stream

### PR DESCRIPTION
**Pull request by sayan133**
Integration test log
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_kinesis_stream []

PRETEST: tests/aws_kinesis_stream

TEST: tests/aws_kinesis_stream
Running terraform
aws_kinesis_stream.named_test_resource: Creating...
aws_kinesis_stream.named_test_resource: Still creating... [10s elapsed]
aws_kinesis_stream.named_test_resource: Still creating... [20s elapsed]
aws_kinesis_stream.named_test_resource: Still creating... [30s elapsed]
aws_kinesis_stream.named_test_resource: Still creating... [40s elapsed]
aws_kinesis_stream.named_test_resource: Creation complete after 41s [id=arn:aws:kinesis:us-east-1:388460667113:stream/turbottest65364]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = "388460667113"
aws_partition = "aws"
aws_region = "us-east-1"
resource_aka = "arn:aws:kinesis:us-east-1:388460667113:stream/turbottest65364"
resource_name = "turbottest65364"

Running SQL query: test-get-query.sql
[
  {
    "account_id": "388460667113",
    "encryption_type": "NONE",
    "key_id": "<null>",
    "partition": "aws",
    "region": "us-east-1",
    "retention_period_hours": 24,
    "stream_arn": "arn:aws:kinesis:us-east-1:388460667113:stream/turbottest65364",
    "stream_name": "turbottest65364",
    "stream_status": "ACTIVE"
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "consumer_count": 0,
    "tags_src": [
      {
        "Key": "foo",
        "Value": "bar"
      }
    ]
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "stream_arn": "arn:aws:kinesis:us-east-1:388460667113:stream/turbottest65364",
    "stream_name": "turbottest65364",
    "stream_status": "ACTIVE"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:kinesis:us-east-1:388460667113:stream/turbottest65364"
    ],
    "tags": {
      "foo": "bar"
    },
    "title": "turbottest65364"
  }
]
✔ PASSED

POSTTEST: tests/aws_kinesis_stream

TEARDOWN: tests/aws_kinesis_stream

SUMMARY:

1/1 passed.

```

### CODE REVIEW

File: aws-test/tests/aws_kinesis_stream/test-get-query.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (Paulami30) [inline aws-test/tests/aws_kinesis_stream/test-get-query.sql:1]: remove space

File: aws/plugin.go [hunk 2]
```diff
@@ -111,6 +111,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"aws_vpc_security_group_rule":            tableAwsVpcSecurityGroupRule(ctx),
 			"aws_vpc_subnet":                         tableAwsVpcSubnet(ctx),
 			"aws_vpc_vpn_gateway":                    tableAwsVpcVpnGateway(ctx),
+			"aws_kinesis_stream":                     tableAwsKinesisStream(ctx),
```
Comment (Paulami30) [inline aws/plugin.go:114]: sort the list

File: aws/table_aws_kinesis_stream.go [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+				Hydrate:     describeStream,
+				Transform:   transform.FromField("StreamDescription.EnhancedMonitoring"),
+			},
+			{
+				Name:        "tags_src",
+				Description: "A list of tags associated with StreamName.",
+				Type:        pb.ColumnType_JSON,
+				Hydrate:     getAwsKinesisStreamTags,
+				Transform:   transform.FromField("Tags"),
+			},
```
Comment (Paulami30) [inline aws/table_aws_kinesis_stream.go:119]:
```suggestion
			},
			
```

File: docs/tables/aws_kinesis_stream.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+select
+	stream_name,
+	stream_arn,
+	stream_status,
+	consumer_count,
+	stream_creation_timestamp,
+	region
+from
+	aws_kinesis_stream;
+```
```
Comment (Paulami30) [inline docs/tables/aws_kinesis_stream.md:19]:
```
select
  stream_name,
  stream_arn,
  stream_status,
  consumer_count,
  stream_creation_timestamp,
  region
from
  aws_kinesis_stream;
```

Comment (Paulami30) [reply docs/tables/aws_kinesis_stream.md:19]: Fix indentation for all the queries

File: docs/tables/aws_kinesis_stream.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+	stream_name,
+	stream_arn,
+	stream_status,
+	consumer_count,
+	stream_creation_timestamp,
+	region
+from
+	aws_kinesis_stream
+where
+	consumer_count = '0';
```
Comment (Paulami30) [inline docs/tables/aws_kinesis_stream.md:52]:
```suggestion
	consumer_count = 0;
```
consumer_count is integer

File: docs/tables/aws_kinesis_stream.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+	stream_arn,
+	encryption_type,
+	key_id,
+	stream_creation_timestamp,
+	region
+from
+	aws_kinesis_stream
+where
+	encryption_type != 'NONE'
+	And key_id = 'alias/aws/kinesis';
```
Comment (Paulami30) [inline docs/tables/aws_kinesis_stream.md:87]:
```suggestion
	and key_id = 'alias/aws/kinesis';
```

File: aws/table_aws_kinesis_stream.go [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+				Type:        pb.ColumnType_JSON,
+				Hydrate:     getAwsKinesisStreamTags,
+				Transform:   transform.FromField("Tags").Transform(kinesisTagListToTurbotTags),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        pb.ColumnType_JSON,
+				Hydrate:     getAwsKinesisStreamAkas,
+				Transform:   transform.FromValue(),
```
Comment (Paulami30) [inline aws/table_aws_kinesis_stream.go:140]: can we get the aka from StreamDescription.StreamARN?

**Review by Paulami30 — DISMISSED** [review]

File: aws/table_aws_kinesis_stream.go [hunk 8]
```diff
@@ -0,0 +1,305 @@
...
+			{
+				Name:        "enhanced_monitoring",
+				Description: "Represents the current enhanced monitoring settings of the stream.",
+				Type:        pb.ColumnType_JSON,
+				Hydrate:     describeStream,
+				Transform:   transform.FromField("StreamDescription.EnhancedMonitoring"),
+			},
+			{
+				Name:        "tags_src",
+				Description: "A list of tags associated with StreamName.",
```
Comment (Subhajit97) [inline aws/table_aws_kinesis_stream.go:114]:
```suggestion
				Description: "A list of tags associated with the stream.",
```

File: aws/table_aws_kinesis_stream.go [hunk 9]
```diff
@@ -0,0 +1,305 @@
...
+	logger.Trace("describeStreamSummary")
+
+	// TODO put me in helper function
+	var region string
+	matrixRegion := plugin.GetMatrixItem(ctx)[matrixKeyRegion]
+	if matrixRegion != nil {
+		region = matrixRegion.(string)
+	}
+	var streamName string
+	streamName = *h.Item.(*kinesis.DescribeStreamOutput).StreamDescription.StreamName
```
Comment (Subhajit97) [inline aws/table_aws_kinesis_stream.go:232]:
```suggestion
	streamName := *h.Item.(*kinesis.DescribeStreamOutput).StreamDescription.StreamName
```

File: aws/table_aws_kinesis_stream.go [hunk 10]
```diff
@@ -0,0 +1,305 @@
...
+
+	return op, nil
+}
+
+//// TRANSFORM FUNCTIONS
+
+func kinesisTagListToTurbotTags(ctx context.Context, d *transform.TransformData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("ssmTagListToTurbotTags")
+	tagList := d.Value.([]*kinesis.Tag)
+
```
Comment (Subhajit97) [inline aws/table_aws_kinesis_stream.go:294]: Return nil, if `tagList` is null
```suggestion

if tagList == nil {
  return nil, nil
}
```

File: docs/tables/aws_kinesis_stream.md [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+  stream_creation_timestamp,
+  region
+from
+  aws_kinesis_stream
+where
+  encryption_type = 'NONE';
+```
+
+
+### List all the encrypted streams which are not encrypted using customer managed key
```
Comment (Subhajit97) [inline docs/tables/aws_kinesis_stream.md:73]:
```suggestion
### List of streams which are not encrypted using CMK
```

File: aws-test/tests/aws_kinesis_stream/variables.json [hunk 12]
```diff
@@ -0,0 +1,3 @@
+{
+  "filter": "resource:{{resourceId}} level:self"
+}
```
Comment (Subhajit97) [inline aws-test/tests/aws_kinesis_stream/variables.json:3]:
```suggestion
```

File: aws-test/tests/aws_kinesis_stream/test-turbot-expected.json [hunk 13]
```diff
@@ -0,0 +1,11 @@
+[
+  {
+    "akas": [
+      "{{ output.resource_aka.value }}"
+    ],
+    "tags": {
+      "foo": "bar"
+    },
+    "title": "{{ output.resource_name.value }}"
```
Comment (Subhajit97) [inline aws-test/tests/aws_kinesis_stream/test-turbot-expected.json:9]:
```suggestion
    "title": "{{ resourceName }}"
```

File: aws-test/tests/aws_kinesis_stream/test-turbot-query.sql [hunk 14]
```diff
@@ -0,0 +1,3 @@
+select akas, tags, title
+from aws.aws_kinesis_stream
+where stream_name = '{{ output.resource_name.value }}';
```
Comment (Subhajit97) [inline aws-test/tests/aws_kinesis_stream/test-turbot-query.sql:3]:
```suggestion
where stream_name = '{{ resourceName }}';
```

File: aws-test/tests/aws_kinesis_stream/test-list-expected.json [hunk 15]
```diff
@@ -0,0 +1,7 @@
+[
+  {
+    "stream_arn": "{{ output.resource_aka.value }}",
+    "stream_name": "{{ output.resource_name.value }}",
```
Comment (Subhajit97) [inline aws-test/tests/aws_kinesis_stream/test-list-expected.json:4]:
```suggestion
    "stream_name": "{{ resourceName }}",
```

File: aws-test/tests/aws_kinesis_stream/test-get-query.sql [hunk 16]
```diff
@@ -0,0 +1,3 @@
+select stream_name, stream_arn, stream_status, encryption_type, key_id, retention_period_hours
```
Comment (Subhajit97) [inline aws-test/tests/aws_kinesis_stream/test-get-query.sql:1]: Also verify `account_id`, `partition` and `region`

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
Make suggested changes, and also add tests logs

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**